src/dsp_tools/utils/project_validate.py

```diff
@@ -13,7 +13,56 @@
 from dsp_tools.utils.excel_to_json_lists import expand_lists_from_excel
 
 
-def check_for_undefined_super_resource(project_definition: dict[str, Any]) -> bool:
+def _check_for_duplicate_names(project_definition: dict[str, Any]) -> bool:
+    """
+    Check that the resource names and property names are unique.
+
+    Args:
+        project_definition: parsed JSON project definition
+
+    Raises:
+        BaseError: detailed error message if there is a duplicate resource name / property name
+
+    Returns:
+        True if the resource/property names are unique
+    """
+    resnames_duplicates: dict[str, set[str]] = dict()
+    propnames_duplicates: dict[str, set[str]] = dict()
+    for onto in project_definition["project"]["ontologies"]:
+        resnames = [r["name"] for r in onto["resources"]]
+        if len(set(resnames)) != len(resnames):
+            for elem in resnames:
+                if resnames.count(elem) > 1:
+                    if not resnames_duplicates.get(onto["name"]):
+                        resnames_duplicates[onto["name"]] = {elem,}
+                    else:
+                        resnames_duplicates[onto["name"]].add(elem)
```

Comment:
```suggestion
                    resnames_duplicates[onto["name"]] = resnames_duplicates.get(onto["name"], set()).add(elem)
```
this could be rewritten to something like this, by using default values on get you don't need to check if it's there or not